framework/src/meshgenerators/ParsedCurveGenerator.C

```diff
@@ -228,6 +236,20 @@ ParsedCurveGenerator::generate()
     elem->subdomain_id() = 1;
   }
 
+  if (_node_set_boundaries.size())
+  {
+    // Add boundary nodesets to boundary info
+    BoundaryInfo & boundary_info = mesh->get_boundary_info();
+    int i = 0;
+    for (auto & side_name : _node_set_boundaries)
+      boundary_info.nodeset_name(i++) = side_name;
```

Comment:
This doesn't seem right. Wouldn't this overwrite possibly unrelated nodeset names? Also use `boundary_id_type` instead of `int`

Reply:
kk